### PR TITLE
Challenge 2 wagmi migration

### PR DESCRIPTION
**Pull request by rin-st**
Challenge 2 migration

Please check if `vercel` works for you. I have error on deployment but it seems something wrong on their side

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,12 +26,14 @@
     "postinstall": "husky install",
     "precommit": "lint-staged",
     "test": "yarn workspace @se-2/hardhat test",
-    "vercel": "yarn workspace @se-2/nextjs vercel",
-    "vercel:yolo": "yarn workspace @se-2/nextjs vercel:yolo"
+    "vercel": "vercel"
```
Comment (Pabl0cks) [inline package.json:29]: I think the problem with `yarn vercel` was originated here, the old vercel scripts dissapeared during the migration.

File: packages/nextjs/pages/events.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ const Events: NextPage = () => {
   const { data: buyTokenEvents, isLoading: isBuyEventsLoading } = useScaffoldEventHistory({
     contractName: "Vendor",
     eventName: "BuyTokens",
-    fromBlock: 0,
+    fromBlock: 0n,
   });
 
   // // SellTokens Events
```
Comment (Pabl0cks) [inline packages/nextjs/pages/events.tsx:16]: I think we need to change the commented `// fromBlock: 0,` for `0n` in SellToken Events also. We'll save some time to the people wondering why their Sell Events are not showing 🙃

**Comment by Pabl0cks** [comment]
> Challenge 2 migration
> 
> Please check if vercel works for you. I have error on deployment but it seems something wrong on their side

Great job @rin-st !! 

It's working great, just added a couple of minor issues to review, one was `yarn vercel`. Will check tomorrow a bit more comparing old deploy vs new deploy, but I don't think I'll find any other issue 🙌

**Review by Pabl0cks — CHANGES_REQUESTED** [review]
> Challenge 2 migration
> 
> Please check if vercel works for you. I have error on deployment but it seems something wrong on their side

Great job @rin-st !! 

It's working great, just added a couple of minor issues to review, one was `yarn vercel`. Will check tomorrow a bit more comparing old deploy vs new deploy, but I don't think I'll find any other issue 🙌

File: package.json (hunk 1 above)
Comment (rin-st) [reply package.json]: Thanks, you're right! Now it works. Confused `package.json`s 😄

File: packages/nextjs/pages/events.tsx (hunk 2 above)
Comment (rin-st) [reply packages/nextjs/pages/events.tsx:16]: thanks, updated

File: package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "postinstall": "husky install",
     "precommit": "lint-staged",
     "test": "yarn workspace @se-2/hardhat test",
-    "vercel": "vercel"
+    "vercel": "yarn workspace @se-2/nextjs vercel"
```
Comment (Pabl0cks) [inline package.json:29]: Should we add back also this? => `"vercel:yolo": "yarn workspace @se-2/nextjs vercel:yolo"`

Comment (rin-st) [reply package.json:29]: no, it was removed from nextjs package https://github.com/scaffold-eth/se-2-challenges/pull/39 , so we don't need it in main package.json too. btw works for other challenges too

Comment (technophile-04) [reply package.json:29]: Yeah 🙌, `yarn vercel` is equivalent to `yarn vercel:yolo` for se-2-challenges repo

I think we forgot to remove it from root package.json in `base-challenge-template`  branch, just created #77 🙌

**Review by technophile-04 — APPROVED** [review]
Looks great !! Tested and everything seems to work fine 🙌 

Tsym Rinat !!

File: package.json (hunk 3 above)
Comment (Pabl0cks) [reply package.json:29]: Ok! Thank you both :) 🙌

**Review by Pabl0cks — APPROVED** [review]

**Comment by technophile-04** [comment]
Tysm all, merging this 🙌 Just updated `useScaffoldEventHistory` hook `hint` to use `n` in README

**Comment by rin-st** [comment]
> Tysm all, merging this 🙌 Just updated useScaffoldEventHistory hook hint to use n in README

Thanks!

**Pull request merged**